### PR TITLE
Change default window mode to "Borderless" on macOS

### PR DESCRIPTION
**Pull request by frenzibyte**
[As mentioned in discord](https://discord.com/channels/188630481301012481/188630652340404224/981699301812101162), using "Fullscreen" takes full control away from the OS, which means that if the game were to deadlock, or if an OS pop-up which freezes the game has opened, or if a breakpoint was hit (happens a lot for me), then the game can't be escaped from and the only option available is to halt the system down using the power button.

I had thought about removing "Fullscreen" entirely on macOS countless times, since it's not friendly in any way, but that prevents the user from running on exclusive fullscreen and/or switching display modes, and I suppose that's probably not fine. Therefore I've kept it for now but changed the default to "Borderless" at least.

### CODE REVIEW

**Comment by peppy** [comment]
If we are keeping "exclusive" mode around, some warning text specific to macOS can probably be added to let users know the implications.

File: osu.Game/OsuGame.cs [hunk 1]
```diff
@@ -658,11 +659,14 @@ protected override void Dispose(bool isDisposing)
...
         protected override IDictionary<FrameworkSetting, object> GetFrameworkConfigDefaults()
-            => new Dictionary<FrameworkSetting, object>
+        {
+            return new Dictionary<FrameworkSetting, object>
             {
-                // General expectation that osu! starts in fullscreen by default (also gives the most predictable performance)
-                { FrameworkSetting.WindowMode, WindowMode.Fullscreen }
+                // General expectation that osu! starts in fullscreen by default (also gives the most predictable performance).
+                // However, macOS is bound to have issues when using exclusive fullscreen as it takes full control away from OS, therefore borderless is default there.
+                { FrameworkSetting.WindowMode, RuntimeInfo.IsApple ? WindowMode.Borderless : WindowMode.Fullscreen }
```
Comment (peppy) [inline osu.Game/OsuGame.cs:667]: `IsApple` includes `iOS` – have you confirmed this doesn't break anything? If not, applying specifically to `RuntimeInfo.OS == Platform.macOS` may be better for safety.

Comment (frenzibyte) [reply osu.Game/OsuGame.cs:667]: Fair enough, iOS runs on `Fullscreen` as its only supported window mode, so probably best to limit the conditional to macOS.

File: osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs [hunk 2]
```diff
@@ -230,6 +231,12 @@ private void updateScreenModeWarning()
                 return;
             }
 
+            if (RuntimeInfo.OS == RuntimeInfo.Platform.macOS)
+            {
+                windowModeDropdown.SetNoticeText(LayoutSettingsStrings.FullscreenMacOSNote, true);
```
Comment (peppy) [inline osu.Game/Overlays/Settings/Sections/Graphics/LayoutSettings.cs:236]: This is pretty contradictory with the message directly above. The game is going to tell the user to run fullscreen, then tell them not to.

I think the whole macOS flow should be separate, without the `NotFullscreenNote` above.

File: osu.Game/Localisation/LayoutSettingsStrings.cs [hunk 3]
```diff
@@ -24,6 +24,11 @@ public static class LayoutSettingsStrings
         /// </summary>
         public static LocalisableString UnableToRunExclusiveFullscreen => new TranslatableString(getKey(@"unable_to_run_exclusive_fullscreen"), @"Unable to run exclusive fullscreen. You'll still experience some input latency.");
 
+        /// <summary>
+        /// "Running fullscreen on macOS takes full control away from the system, this could lead to system-wide freezes if the game becomes unresponsive."
+        /// </summary>
+        public static LocalisableString FullscreenMacOSNote => new TranslatableString(getKey(@"fullscreen_macos_note"), @"Running fullscreen on macOS takes full control away from the system, this could lead to system-wide freezes if the game becomes unresponsive.");
```
Comment (peppy) [inline osu.Game/Localisation/LayoutSettingsStrings.cs:30]: How about something like

`Using fullscreen on macOS makes interacting with the menu bar and spaces no longer work, and may lead to freezes if a system dialog is presented. Using borderless is recommended.`

**Review by peppy — CHANGES_REQUESTED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**